tests/conftest.py

```diff
@@ -238,3 +239,38 @@ def unprivileged_client(
 
         else:
             yield admin_client
+
+
+@pytest.fixture(scope="session")
+def dsc_resource(admin_client: DynamicClient, name: str = "default-dsc"):
+    for dsc in DataScienceCluster.get(dyn_client=admin_client, name=name):
+        yield dsc
+
+
+def get_dsc_component_state(dsc_resource: DataScienceCluster, component_name: str) -> str:
```

Comment:
used once and it is just wrapping the access to the resource `managementState` . use directly where needed